skrub/_table_vectorizer.py

```diff
@@ -1125,28 +958,54 @@ def get_feature_names_out(self, input_features=None) -> list[str]:
         list of str
             Feature names.
         """
-        ct_feature_names = super().get_feature_names_out()
-        all_trans_feature_names = []
+        return self._column_transformer.get_feature_names_out(input_features)
 
-        for name, trans, cols, _ in self._iter(
-            fitted=True, replace_strings=False, column_as_strings=False
+    @property
+    def transformers_(self):
+        """Transformers applied to the different columns."""
+        # For the "remainder" columns, the `ColumnTransformer` `transformers_`
+        # attribute contains the index instead of the column name,
+        # so we convert the values to the appropriate column names
+        # if there is less than 20 columns in the remainder.
+        transformers = []
+        for i, (name, transformer, columns) in enumerate(
+            self._column_transformer.transformers_
         ):
-            if isinstance(trans, str):
-                if trans == "drop":
-                    continue
-                elif trans == "passthrough":
-                    if all(isinstance(col, int) for col in cols):
-                        cols = [self.columns_[i] for i in cols]
-                    all_trans_feature_names.extend(cols)
-                continue
-            trans_feature_names = trans.get_feature_names_out(cols)
-            all_trans_feature_names.extend(trans_feature_names)
-
-        if len(ct_feature_names) != len(all_trans_feature_names):
-            warn("Could not extract clean feature names; returning defaults. ")
-            return list(ct_feature_names)
-
-        return all_trans_feature_names
+            if name == "remainder" and len(columns) < 20:
```

Comment:
Let's add a TODO to remove this when https://github.com/scikit-learn/scikit-learn/issues/27533 is closed.